elastic/apm-server
# processor/otel: set Span.RepresentativeCount

## Motivation/summary

If `sampler.type` and `sampler.param` attributes are
available, use them to set `Span.RepresentativeCount`
as well as for transactions. This will enable us to
calculate edge metrics for Jaeger.

## Checklist

~- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~

I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

1. Instrument an app with Jaeger (dummy program is fine), with some outgoing HTTP spans (i.e. which will have destination service resource set)
2. Configure Jaeger with probabilistic sampling, set the param to 0.5.
3. Ensure that edge metrics are produced

## Related issues

Closes #4113
